sourcery_analytics/extractors.py

```diff
@@ -159,6 +167,29 @@ def _extract_from_directory(self, directory: pathlib.Path) -> typing.Iterator[T]
         )
 
     def _extract_from_file(self, file: pathlib.Path) -> typing.Iterator[T]:
-        manager = astroid.manager.AstroidManager()
-        module = manager.ast_from_file(file)
-        yield from self._extract_from_node(module)
+        try:
+            module = self.manager.ast_from_file(file)
+            yield from self._extract_from_node(module)
+        except astroid.AstroidSyntaxError as e:
+            if isinstance(e.error, SyntaxError):
+                error_message = _format_syntax_error_message(
+                    "skipping file", file, e.error
+                )
+            else:
+                error_message = str(e).replace("\n", " ")
+            warning = SyntaxWarning(error_message)
+            warnings.warn(warning)
+            yield from ()
```

Comment:
**suggestion (edge_case_not_handled):** Ensure proper handling of all exceptions during file parsing.

While handling `AstroidSyntaxError` is crucial, consider also catching and handling other potential exceptions that might arise during file parsing, such as IOErrors. This ensures robust error handling and improves the resilience of the extraction process.

```suggestion
        except astroid.AstroidSyntaxError as e:
            if isinstance(e.error, SyntaxError):
                error_message = _format_syntax_error_message(
                    "skipping file", file, e.error
                )
            else:
                error_message = str(e).replace("\n", " ")
            warning = SyntaxWarning(error_message)
            warnings.warn(warning)
            yield from ()
        except IOError as e:
            error_message = f"IOError encountered when processing file {file}: {e}"
            warning = SyntaxWarning(error_message)
            warnings.warn(warning)
            yield from ()
```